fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/MvUtils.java

```diff
@@ -1455,4 +1469,70 @@ public static SlotRef extractPartitionSlotRef(Expr paritionExpr) {
+     * modified columns have done because the modified process may be failed and in this situation
+     * should not inactive mvs then.
+     */
+    public static void inactiveRelatedMaterializedViews(Database db,
+                                                        OlapTable olapTable,
+                                                        Set<String> modifiedColumns) {
+        if (modifiedColumns == null || modifiedColumns.isEmpty()) {
+            return;
+        }
+        // inactive related asynchronous mvs
+        for (MvId mvId : olapTable.getRelatedMaterializedViews()) {
+            MaterializedView mv = (MaterializedView) db.getTable(mvId.getId());
+            if (mv == null) {
+                LOG.warn("Ignore materialized view {} does not exists", mvId);
+                continue;
+
+            }
+            // TODO: support more types for base table's schema change.
+            try {
+                MvPlanContext mvPlanContext = MvPlanContextBuilder.getPlanContext(mv);
+                if (mvPlanContext != null) {
+                    OptExpression mvPlan = mvPlanContext.getLogicalPlan();
+                    List<ColumnRefOperator> usedColRefs = MvUtils.collectScanColumn(mvPlan, scan -> {
+                        if (scan == null) {
+                            return false;
+                        }
+                        Table table = scan.getTable();
+                        return table.getId() == olapTable.getId();
+                    });
+                    Set<String> usedColNames = usedColRefs.stream().map(x -> x.getName()).collect(Collectors.toSet());
+                    for (String modifiedColumn : modifiedColumns) {
+                        if (usedColNames.contains(modifiedColumn)) {
+                            LOG.warn("Setting the materialized view {}({}) to invalid because " +
+                                            "the column {} of the table {} was modified.", mv.getName(), mv.getId(),
+                                    modifiedColumn, olapTable.getName());
+                            mv.setInactiveAndReason(
+                                    "base-table schema changed for columns: " + Joiner.on(",").join(modifiedColumns));
+                        }
+                    }
+                }
+            } catch (SemanticException e) {
+                LOG.warn("Get related materialized view {} failed:", mv.getName(), e);
+                LOG.warn("Setting the materialized view {}({}) to invalid because " +
+                                "the columns  of the table {} was modified.", mv.getName(), mv.getId(),
+                        olapTable.getName());
+                mv.setInactiveAndReason(
+                        "base-table schema changed for columns: " + Joiner.on(",").join(modifiedColumns));
+            } catch (Exception e) {
+                LOG.warn("Get related materialized view {} failed:", mv.getName(), e);
```

Comment:
why?

Reply:
There may be a lot of reasons why cause expectations: optimization timeout, rule bugs...